### PR TITLE
Testing pre-commit CI

### PR DESCRIPTION
**Pull request by alexnguyenn**
A bad PR to test whether pre-commit CI works properly

### CODE REVIEW

**Comment by alexnguyenn** [comment]
Pinging @gwwatkin

**Comment by gwwatkin** [comment]
@alexnguyenn looks like it works! No config in the repo was needed?

**Comment by alexnguyenn** [comment]
Nope, same developer with `pre-commit`, so we need is [`.pre-commit-config.yaml`](https://github.com/latticesurgery-com/lattice-surgery-compiler/blob/dev/.pre-commit-config.yaml)

**Comment by gwwatkin** [comment]
Can you maybe add a few words on how you did it to this page? https://github.com/latticesurgery-com/lattice-surgery-compiler/wiki/Unitary-Fund-Python-project-set-up-guide

**Comment by alexnguyenn** [comment]
I will take a look at it during the weekend

**Comment by alexnguyenn** [comment]
pre-commit.ci autofix

File: src/lsqecc/pipeline/lattice_surgery_compilation_pipeline.py [hunk 1]
```diff
@@ -14,16 +14,15 @@
...
-
 from typing import List, Optional, Tuple
 
 import qiskit.visualization as qkvis
 from qiskit import circuit as qkcirc
 
+import lsqecc.lattice_array.visual_array_cell as vac
+import lsqecc.logical_lattice_ops.logical_lattice_ops as llops
 import lsqecc.patches.lattice_surgery_computation_composer as lscc
+import lsqecc.pauli_rotations.segmented_qasm_parser as segmented_qasm_parser
```
Comment (gwwatkin) [inline src/lsqecc/pipeline/lattice_surgery_compilation_pipeline.py]: Wonder why it made these changes

File: src/lsqecc/simulation/qiskit_opflow_utils.py [hunk 2]
```diff
@@ -15,11 +15,11 @@
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301
 # USA
 
-from typing import Dict, List, Optional
+import qiskit.quantum_info as qkinfo
 
 import qiskit.aqua.operators as qk
-import qiskit.exceptions as qkexcept
-import qiskit.quantum_info as qkinfo
+from qiskit.quantum_info import *
```
Comment (gwwatkin) [inline src/lsqecc/simulation/qiskit_opflow_utils.py]: Looks like `pre-commit.ci` si not clever enough to figure this change. Fair enough, though. The occurrences of `qkinfo` in the code are lost - I'd say it's beyond the scope of an import linter to catch it.

Maybe worth testing again without this `*` change?

File: src/lsqecc/pipeline/lattice_surgery_compilation_pipeline.py (hunk 1 above)
Comment (alexnguyenn) [reply src/lsqecc/pipeline/lattice_surgery_compilation_pipeline.py]: I am pretty sure it was `isort` - which sorts all of your imports.

File: src/lsqecc/simulation/qiskit_opflow_utils.py (hunk 2 above)
Comment (alexnguyenn) [reply src/lsqecc/simulation/qiskit_opflow_utils.py]: I don't think `isort` will fix `import *` issues, but from the [run report](https://results.pre-commit.ci/run/github/298419377/1636684482.zlAoKkjETNmgeHtCpuO-Tg) - can be access by clicking Details, `flake8` catches it which failed the checks.

Comment (alexnguyenn) [reply src/lsqecc/simulation/qiskit_opflow_utils.py]: oh if you are talking about `qkinfo`, isort just moved it to a different position.

Comment (gwwatkin) [reply src/lsqecc/simulation/qiskit_opflow_utils.py]: ... I see, are you going to try it again with a different issue?

Comment (gwwatkin) [reply src/lsqecc/simulation/qiskit_opflow_utils.py]: I meant to say that `pre-commit.ci autofix` dosen't seem to fix many things, even after running it, the CI has errors

Comment (gwwatkin) [reply src/lsqecc/simulation/qiskit_opflow_utils.py]: NVM I get it, I thought black and isort were supposed to revert the change you introduced.

Comment (alexnguyenn) [reply src/lsqecc/simulation/qiskit_opflow_utils.py]: Yep as I mentioned in https://github.com/latticesurgery-com/lattice-surgery-compiler/pull/147#issuecomment-966790546, `pre-commit.ci autofix` will only push changes made by black and isort which does not guarantee that errors flagged by other hooks (mainly flake8 and mypy) will be fixed.

**Pull request closed without merging**